scipy/stats/tests/test_distributions.py

```diff
@@ -5235,6 +5235,14 @@ def test_fitstart_valid(self):
             k, df, _, _ = stats.studentized_range._fitstart([1, 2, 3])
         assert_(stats.studentized_range._argcheck(k, df))
 
+    def test_clipping(self):
+        # The result of this computation was -9.9253938401489e-14 on some
+        # systems, but p-values should not be negative. The result from mpmath
+        # is 0.
+        q, k, v = 34.6413996195345746, 3, 339
+        p = stats.studentized_range.sf(q, k, v)
+        assert_equal(p >= 0, True)
```

Comment:
```suggestion
        assert_allclose(p, 0, atol=1e-10)
        assert p >= 0 
```

Reply:
See my comments that were hidden by the latest commit.
https://github.com/scipy/scipy/pull/15525/commits/175f5fcc131bded726e2f81a866b8d9396b71a67